jaxlib/lapack.pyx

```diff
@@ -1497,7 +1499,7 @@ cdef void _unpack_float_eigenvectors(
   cdef int j, k
   j = 0
   while j < n:
-    if im_eigenvalues[j] == 0.:
+    if im_eigenvalues[j] == 0. or im_eigenvalues[j] != im_eigenvalues[j]:
```

Comment:
Use `isnan` instead, to make this a little more self-evident?

from https://stackoverflow.com/questions/8200311/checking-for-nan-in-cython
`from libc.math cimport isnan`